include/flox/core/exceptions.hh

```diff
@@ -74,76 +74,150 @@ enum error_category {
 /** Typed exception wrapper used for misc errors. */
 class FloxException : public std::exception
 {
-private:
 
-  /** Corresponds to an error_category, in this case `EC_FLOX_EXCEPTION`. */
-  static constexpr std::string_view categoryMsg = "general error";
+private:
 
   /** Additional context added when the error is thrown. */
   std::optional<std::string> contextMsg;
 
   /**
-   * If some other exception was caught before throwing this one, caughtMsg
+   * If some other exception was caught before throwing this one, @a caughtMsg
    * contains what() of that exception.
    */
   std::optional<std::string> caughtMsg;
 
+  /** The final what() message. */
+  std::string whatMsg;
+
+
 public:
 
   explicit FloxException( std::string_view contextMsg )
     : contextMsg( contextMsg )
-  {}
+  {
+    this->whatMsg = std::string( this->getCategoryMessage() ) + ": "
+                    + std::string( contextMsg );
```

Comment:
nit: you are inconsistent in putting `whatMsg( categoryMsg )` in the initializer list and then whether you use `+ std::string()`

Reply:
Whoops this one should have been removed. Good catch